### PR TITLE
Split up HBRequestContext from HBRequest

### PR DESCRIPTION
**Pull request by Joannis**

### CODE REVIEW

**Comment by tib** [comment]
Can't wait to try this out @Joannis 👍

File: Sources/Hummingbird/Server/Request.swift [hunk 1]
```diff
@@ -50,23 +50,9 @@ public struct HBRequest: Sendable, HBSendableExtensible {
...
-    /// EventLoop request is running on
-    public var eventLoop: EventLoop { self._internal.context.eventLoop }
-    /// ByteBuffer allocator used by request
-    public var allocator: ByteBufferAllocator { self._internal.context.allocator }
-    /// IP request came from
-    public var remoteAddress: SocketAddress? { self._internal.context.remoteAddress }
-
-    /// Request ID (Uses ID attached to loggger to avoid replication)
-    public var id: String { self.logger[metadataKey: "hb_id"]!.description }
+    public let id: String
```
Comment (adam-fowler) [inline Sources/Hummingbird/Server/Request.swift:55]: Is this something we can move to context as well? I was kind of hoping we could ditch `HBRequest` for `HBHTTPRequest`. The ID was something you asked for initially what was the intention for it.

Comment (adam-fowler) [reply Sources/Hummingbird/Server/Request.swift:55]: In fact the id is stored in both request and context at the moment

File: Sources/Hummingbird/Server/Request.swift [hunk 2]
```diff
@@ -133,66 +114,19 @@ public struct HBRequest: Sendable, HBSendableExtensible {
...
     final class _Internal: Sendable {
-        internal init(uri: HBURL, version: HTTPVersion, method: HTTPMethod, headers: HTTPHeaders, applicationContext: HBApplication.Context, context: HBRequestContext, endpointPath: String? = nil) {
+        internal init(uri: HBURL, version: HTTPVersion, method: HTTPMethod, headers: HTTPHeaders, endpointPath: String? = nil) {
             self.uri = uri
             self.version = version
             self.method = method
             self.headers = headers
-            self.applicationContext = applicationContext
-            self.context = context
             self.endpointPath = .init(endpointPath)
```
Comment (adam-fowler) [inline Sources/Hummingbird/Server/Request.swift:130]: Similar to id any reason we can't move endpointPath to the context as well

**Review by adam-fowler — COMMENTED** [review]
Looks good in general. Couple of question surrounding why a couple of things are still in HBRequest and not the HBRequestContext.

**Review by adam-fowler — APPROVED** [review]
LGTM

**Pull request merged**